dashboard/test/models/course_offering_test.rb

```diff
@@ -300,6 +300,79 @@ class CourseOfferingTest < ActiveSupport::TestCase
     refute unit1.course_version.course_offering.any_versions_in_development?
   end
 
+  test 'any_version_is_in_published_state? is false if none of the course versions have a published_state of preview or stable' do
+    unit1 = create(:script, name: 'unit1', family_name: 'family-10', version_year: '1991', is_course: true, published_state: 'beta')
+    CourseOffering.add_course_offering(unit1)
+    unit2 = create(:script, name: 'unit2', family_name: 'family-10', version_year: '1992', is_course: true, published_state: 'beta')
+    CourseOffering.add_course_offering(unit2)
+
+    refute unit1.course_version.course_offering.any_version_is_in_published_state?
+  end
+
+  test 'any_version_is_in_published_state? is true if one of the course versions have a published_state of preview or stable' do
+    unit1 = create(:script, name: 'unit1', family_name: 'family-11', version_year: '1991', is_course: true, published_state: 'beta')
+    CourseOffering.add_course_offering(unit1)
+    unit2 = create(:script, name: 'unit2', family_name: 'family-11', version_year: '1992', is_course: true, published_state: 'preview')
+    CourseOffering.add_course_offering(unit2)
+
+    assert unit1.course_version.course_offering.any_version_is_in_published_state?
+  end
+
+  test 'any_version_is_in_published_state? is true if all of the course versions have a published_state of preview or stable' do
+    unit1 = create(:script, name: 'unit1', family_name: 'family-12', version_year: '1991', is_course: true, published_state: 'stable')
+    CourseOffering.add_course_offering(unit1)
+    unit2 = create(:script, name: 'unit2', family_name: 'family-12', version_year: '1992', is_course: true, published_state: 'preview')
+    CourseOffering.add_course_offering(unit2)
+
+    assert unit1.course_version.course_offering.any_version_is_in_published_state?
+  end
+
+  test 'assignable_published_for_students_course_offerings filters only for assignable, published, and for student course offerings' do
+    # Course offering that doesn't satisfy any of the conditions
+    none_unit = create(:script, name: 'unit1', family_name: 'none', version_year: '1991', is_course: true, published_state: 'in_development', instructor_audience: 'universal_instructor', participant_audience: 'teacher')
+    none_co = CourseOffering.add_course_offering(none_unit)
+    none_co.assignable = false
+    none_co.save!
```

Comment:
nit: please shorten this pattern, here and throughout this test case
```suggestion
    none_co.update!(assignable: false)
```